client/src/components/Input/TextChat.jsx

```diff
@@ -1,10 +1,11 @@
-import React, { useEffect, useContext, useRef } from 'react';
+import React, { useEffect, useContext, useRef, useState } from 'react';
 import TextareaAutosize from 'react-textarea-autosize';
 import { useRecoilValue, useRecoilState, useSetRecoilState } from 'recoil';
 import SubmitButton from './SubmitButton';
 import OptionsBar from './OptionsBar';
 import { EndpointMenu } from './EndpointMenu';
 import Footer from './Footer';
+import useSpeechRecognition from './SpeechRecognition';
```

Comment:
you should put the hooks in their respective directory for uniformity and organization, name them after the function 'use...'